client startup time regression

While testing some other work, I noticed that we seem to have a regression in client startup speed. There's a roughly 8 second delay between `client.fingerprint_mgr.network` and `client.plugin` log messages, and the CLI UI hides all logs during that window so it just looks like it's hanging. Not a critical bug but something we should address when we have the time.

I noticed this on `main` but I've tested this on every version of 1.5.x, 1.4.x, and on 1.3.12:

```
==> Loaded configuration from /home/tim/ws/nomad/etc/local/client.hcl
==> Starting Nomad agent...
==> Nomad agent configuration:

       Advertise Addrs: HTTP: 192.168.1.239:5646
            Bind Addrs: HTTP: [0.0.0.0:5646]
                Client: true
             Log Level: info
                Region: global (DC: dc1)
                Server: false
               Version: 1.3.12

==> Nomad agent started! Log data will stream in below:

    2023-03-30T11:12:30.571-0400 [ERROR] agent.plugin_loader.docker: failed to list pause containers: plugin_dir=/opt/nomad/plugins error=<nil>
    2023-03-30T11:12:30.572-0400 [ERROR] agent.plugin_loader.docker: failed to list pause containers: plugin_dir=/opt/nomad/plugins error=<nil>
    2023-03-30T11:12:30.572-0400 [INFO]  agent: detected plugin: name=java type=driver plugin_version=0.1.0
    2023-03-30T11:12:30.572-0400 [INFO]  agent: detected plugin: name=docker type=driver plugin_version=0.1.0
    2023-03-30T11:12:30.572-0400 [INFO]  agent: detected plugin: name=raw_exec type=driver plugin_version=0.1.0
    2023-03-30T11:12:30.572-0400 [INFO]  agent: detected plugin: name=exec type=driver plugin_version=0.1.0
    2023-03-30T11:12:30.572-0400 [INFO]  agent: detected plugin: name=qemu type=driver plugin_version=0.1.0
    2023-03-30T11:12:30.572-0400 [INFO]  client: using state directory: state_dir=/run/nomad/client/data/client
    2023-03-30T11:12:30.572-0400 [INFO]  client: using alloc directory: alloc_dir=/run/nomad/client/data/alloc
    2023-03-30T11:12:30.572-0400 [INFO]  client: using dynamic ports: min=20000 max=32000 reserved=""
    2023-03-30T11:12:30.599-0400 [INFO]  client.fingerprint_mgr.cgroup: cgroups are available
    2023-03-30T11:12:30.605-0400 [INFO]  client.fingerprint_mgr.consul: consul agent is available
    2023-03-30T11:12:30.634-0400 [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=wlp0s20f3
    2023-03-30T11:12:30.635-0400 [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=lo
    2023-03-30T11:12:30.640-0400 [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=wlp0s20f3
    2023-03-30T11:12:30.645-0400 [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=virbr0
    2023-03-30T11:12:30.648-0400 [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=enx9cebe891305a
    2023-03-30T11:12:30.650-0400 [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=docker0
    2023-03-30T11:12:38.817-0400 [INFO]  client.plugin: starting plugin manager: plugin-type=csi
    2023-03-30T11:12:38.817-0400 [INFO]  client.plugin: starting plugin manager: plugin-type=driver
    2023-03-30T11:12:38.818-0400 [INFO]  client.plugin: starting plugin manager: plugin-type=device
    2023-03-30T11:12:38.823-0400 [INFO]  client.consul: discovered following servers: servers=[192.168.1.239:4647]
    2023-03-30T11:12:38.851-0400 [INFO]  client: started client: node_id=e9761c32-4800-ca59-95d9-052a6ff91f93
    2023-03-30T11:12:38.852-0400 [WARN]  http: enable_debug is set to true. This is insecure and should not be enabled in production
    2023-03-30T11:12:38.852-0400 [WARN]  http: enable_debug is set to true. This is insecure and should not be enabled in production
    2023-03-30T11:12:38.852-0400 [INFO]  agent.joiner: starting retry join: servers=127.0.0.1:4647
    2023-03-30T11:12:38.852-0400 [INFO]  client: node registration complete
    2023-03-30T11:12:38.853-0400 [INFO]  agent.joiner: retry join completed: initial_servers=1 agent_mode=client
    2023-03-30T11:12:45.858-0400 [INFO]  client: node registration complete
```

--- a/website/content/docs/configuration/client.mdx
+++ b/website/content/docs/configuration/client.mdx
@@ -344,12 +344,14 @@ see the [drivers documentation](/nomad/docs/drivers).
 
 - `"fingerprint.denylist"` `(string: "")` - Specifies a comma-separated list of
   denylisted fingerprinters. If specified, any fingerprinters in the denylist
-  will be disabled.
+  will be disabled. A common use-case for the fingerprint denylist is to disable
+  fingerprinters of irrelevant cloud environments, which can slow down client
+  agent startup time.
 
   ```hcl
   client {
     options = {
-      "fingerprint.denylist" = "network"
+      "fingerprint.denylist" = "env_aws,env_gce,env_azure,env_digitalocean"
     }
   }
   ```

